cmd/frontend/graphqlbackend/external_services.go

```diff
@@ -444,3 +447,51 @@ func (r *schemaResolver) CancelExternalServiceSync(ctx context.Context, args *ca
 
 	return &EmptyResponse{}, nil
 }
+
+type externalServiceNamespacesArgs struct {
+	Kind  string
+	Token string
+	Url   string
+}
+
+func (r *schemaResolver) QueryExternalServiceNamespaces(ctx context.Context, args *externalServiceNamespacesArgs) (*queryExternalServiceNamespaceConnectionResolver, error) {
+	if auth.CheckCurrentUserIsSiteAdmin(ctx, r.db) != nil {
+		return nil, auth.ErrMustBeSiteAdmin
+	}
+
+	res := queryExternalServiceNamespaceConnectionResolver{
+		args:              args,
+		repoupdaterClient: r.repoupdaterClient,
+	}
+	return &res, nil
+}
+
+type queryExternalServiceNamespaceConnectionResolver struct {
+	args              *externalServiceNamespacesArgs
+	repoupdaterClient *repoupdater.Client
+
+	once       sync.Once
+	nodes      []*types.ExternalServiceNamespace
+	totalCount int32
+	err        error
+}
+
+func NewSourceConnection(kind, url, token string) (string, error) {
+	var (
+		err        error
+		marshalled []byte
+	)
+	switch kind {
+	case extsvc.KindGitHub:
+		cnxn := schema.GitHubConnection{
+			Url:   url,
+			Token: token,
+		}
+
+		marshalled, err = jsoniter.Marshal(cnxn)
+	default:
+		return "", errors.New(repos.UnimplementedDiscoverySource)
+	}
+
+	return string(marshalled), err
+}
```

Comment:
Could be simplified to:

```suggestion
func NewSourceConnection(kind, url, token string) (string, error) {
	switch kind {
	case extsvc.KindGitHub:
		cnxn := schema.GitHubConnection{
			Url:   url,
			Token: token,
		}
		marshalled, err := jsoniter.Marshal(cnxn)
		return string(marshalled), err
	default:
		return "", errors.New(repos.UnimplementedDiscoverySource)
	}
}
```